ui/src/pages/Dashboard/DashboardModal.tsx

```diff
@@ -0,0 +1,192 @@
+            } else if (activeState === 'no') {
+                extractColumnsValues = mergeInputValues(inactiveInputs, activeInputs);
+            }
+        } else if (selectTitleForDropdownInModal === 'Account') {
+            extractColumnsValues = processResults(values[1].results, selectTitleForDropdownInModal);
+        } else {
+            extractColumnsValues = processResults(values[2].results, selectTitleForDropdownInModal);
+        }
+
+        setDataIngestionDropdownValues(extractColumnsValues);
+
+        const eventsQuery = createNewQueryForNumberOfEventsInModal(
+            selectTitleForDropdownInModal,
+            selectValueForDropdownInModal,
+            copyDataIngestionModalJson.dataSources.ds_search_1.options.query
+        );
+        const dataVolumeQuery = createNewQueryForDataVolumeInModal(
+            selectTitleForDropdownInModal,
+            selectValueForDropdownInModal,
+            copyDataIngestionModalJson.dataSources.data_ingestion_modal_data_volume_ds.options.query
+        );
+
+        copyDataIngestionModalJson.dataSources.data_ingestion_modal_data_volume_ds.options.query =
+            dataVolumeQuery;
+        copyDataIngestionModalJson.dataSources.ds_search_1.options.query = eventsQuery;
+
+        // Modify visualizations only for specific cases
+        if (
+            selectTitleForDropdownInModal === 'Input' ||
+            selectTitleForDropdownInModal === 'Account'
+        ) {
+            // Remove data volume visualization for "Input" and "Account"
+            delete copyDataIngestionModalJson.visualizations.data_ingestion_modal_data_volume_viz;
+            delete copyDataIngestionModalJson.dataSources.data_ingestion_modal_data_volume_ds;
+            delete copyDataIngestionModalJson.layout.structure[2];
+            copyDataIngestionModalJson.layout.structure =
+                copyDataIngestionModalJson.layout.structure.filter(
+                    (item: Record<string, unknown>) => item !== null
+                );
+            copyDataIngestionModalJson.layout.structure[2].position.y = 80;
+        } else if (selectTitleForDropdownInModal === 'Host') {
+            // Remove event count visualization for "Host"
+            delete copyDataIngestionModalJson.visualizations.data_ingestion_modal_events_count_viz;
+            delete copyDataIngestionModalJson.dataSources.ds_search_1;
+            delete copyDataIngestionModalJson.layout.structure[3];
+            copyDataIngestionModalJson.layout.structure =
+                copyDataIngestionModalJson.layout.structure.filter(
+                    (item: Record<string, unknown>) => item !== null
+                );
+        }
+
+        return copyDataIngestionModalJson;
+    }, [
+        dataIngestionModalDef,
+        globalConfig,
+        selectTitleForDropdownInModal,
+        selectValueForDropdownInModal,
+        setDataIngestionDropdownValues,
+    ]);
```

Comment:
This callback needs to be change as currently we make a call in line 
```
        const values = await fetchDropdownValuesFromQuery(globalConfig);
```
to get the data which we only need to load once
Currently every time you change dropdown selection there are 3 search jobs done